apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Span.java

```diff
@@ -234,6 +243,20 @@ public void beforeEnd(long epochMicros) {
         if (type == null) {
             type = "custom";
         }
+
+        // set outcome when not explicitly set by user nor instrumentation
+        if (outcomeNotSet()) {
+            Outcome outcome = Outcome.UNKNOWN;
```

Comment:
implementation shortcut to handle all the most common cases at once:
- HTTP spans from status code
- DB, messaging and other known sub-types where any thrown exception would mean span failure.